app/src/main/java/com/li/weatherapp/data/source/remote/NewsRemoteDataSource.kt

```diff
@@ -0,0 +1,84 @@
+package com.li.weatherapp.data.source.remote
+
+import com.li.weatherapp.BuildConfig
+import com.li.weatherapp.data.model.News
+import com.li.weatherapp.data.source.NewsDataSource
+import com.li.weatherapp.data.source.remote.utils.RemoteAsyncTask
+import com.li.weatherapp.data.source.remote.utils.makeNetworkCall
+import com.li.weatherapp.data.source.utils.OnDataLoadCallback
+import org.w3c.dom.Document
+import org.w3c.dom.NodeList
+import org.xml.sax.InputSource
+import java.io.StringReader
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+import javax.xml.parsers.DocumentBuilderFactory
+import javax.xml.parsers.ParserConfigurationException
+
+@Suppress("DEPRECATION")
+class NewsRemoteDataSource private constructor() : NewsDataSource {
+
+    override fun getNews(callback: OnDataLoadCallback<List<News>>) {
+        RemoteAsyncTask(callback) {
+            getNews()
+        }.execute()
+    }
+
+    private fun getDocumentFromXml(xml: String): Document? {
+        val document: Document
+        val factory = DocumentBuilderFactory.newInstance()
+        try {
+            val documentBuilder = factory.newDocumentBuilder()
+            val inputSource = InputSource()
+            inputSource.characterStream = StringReader(xml)
+            inputSource.encoding = ENCODING
```

Comment:
dùng scope function tránh lặp code 
```suggestion
            val inputSource = InputSource().apply {
               characterStream = StringReader(xml)
               encoding = ENCODING
            }
```